lib/spack/spack/test/cmd/install.py

```diff
@@ -1133,3 +1133,9 @@ def test_invalid_concurrent_packages_flag(mutable_config):
     install = SpackCommand("install")
     with pytest.raises(ValueError, match="expected a positive integer"):
         install("--concurrent-packages", "-2", fail_on_error=False)
+
+
+def test_concurrent_packages_set_in_config(mutable_config):
+    """Ensure that the number of concurrent packages is properly set from adding to config"""
+    spack.config.set("config:concurrent_packages", 3)
+    assert spack.config.get("config:concurrent_packages") == 3
```

Comment:
I think it's better to construct the PackageInstaller object and check there, to ensure the value is passed through properly

```suggestion
    spec = spack.concretize.concretize_one("pkg-a")
    installer = spack.installer.PackageInstaller([spec.package])
    assert installer.concurrent_packages == 3
```